staging/src/k8s.io/client-go/tools/cache/listwatch.go

```diff
@@ -49,9 +49,11 @@ type WatchFunc func(options metav1.ListOptions) (watch.Interface, error)
 // It is a convenience function for users of NewReflector, etc.
 // ListFunc and WatchFunc must not be nil
 type ListWatch struct {
-	ListFunc      ListFunc
-	WatchFunc     WatchFunc
-	DisablePaging bool
+	ListFunc  ListFunc
+	WatchFunc WatchFunc
+	// DisableChunking requests no chunking for this list watcher. It has no effect in Kubernetes 1.8, but in
+	// 1.9 will allow a controller to opt out of chunking.
+	DisableChunking bool
```

Comment:
can this be named in the positive, especially since that would get the default behavior you want?

Reply:
The plan was in 1.9 to remove the `false &&` below.  All of the test suites are already configured to bypass chunking by setting this flag, whereas all clients would opt in for beta.  I wanted to avoid pointer insanity in internal code as well as bad globals (since this is core library code).